source/globalCommands.py

```diff
@@ -3300,6 +3300,7 @@ def script_braille_toggleShowCursor(self, gesture):
 			state = _("Braille cursor on")
 			config.conf["braille"]["showCursor"]=True
 		ui.message(state)
+		braille.handler.update()

```

Comment:
Could you explain in a code comment why this is necessary? I'm also a bit concerned by the fact that you're updating the braille handler after executing ui.message, which should print a message to the braille display as well.

Reply:
I have not noticed problems with ui.message and braille.handler.update.

But unfortunately there is another problem: at least with Albatross when I type "dir/s" in NVDA directory whic contains source directory for example, result is that it takes long time when all text is brailled (34842 File(s) and 11846 Dir(s)).

There should be for example a queue with suitable length to avoid the situation that all thousands of handleUpdates should be performed.

Do you have approach you prefer?

Reply:
Sure, I actually expected something like that. I think you can merge in the changes in braille.py for #15156 to compensate for this delay, though we might also need to move the progress bar logic out of handleUpdate.